spanner/value.go

```diff
@@ -3564,6 +3611,21 @@ func encodeValue(v interface{}) (*proto3.Value, *sppb.Type, error) {
 		pt = proto.Clone(v.Type).(*sppb.Type)
 	case []GenericColumnValue:
 		return nil, nil, errEncoderUnsupportedType(v)
+	case protoreflect.Enum:
+		// TODO: here nil check not needed because a enum variable always has a default value. Recheck
```

Comment:
Is this TODO being looked into?

Reply:
I have removed the TODO comment. As we are fetching `.Number()`, it would be good to have the nil check.